app/src/main/java/mok/it/app/mokapp/feature/project_add_participants/AddOrRemoveMembersFragment.kt

```diff
@@ -0,0 +1,217 @@
+                    }) {
+                    Text(text = "Mentés")
+                }
+            },
+        ) { padding ->
+            Column(
+                modifier = Modifier.padding(bottom = padding.calculateBottomPadding())
+            ) {
+                when (showDialog) {
+                    DialogType.CONFIRM -> {
+                        OkCancelDialog(
+                            text = "Biztos, hogy el szeretnéd menteni a módosításokat? Ha valakit törölsz a projektből, az elveszíti az összes eddig megszerzett mancsát rajta.",
+                            onConfirm = {
+                                viewModel.updateMembersOfProject()
+                                Toast.makeText(
+                                    context,
+                                    "Résztvevők listája módosítva!",
+                                    Toast.LENGTH_SHORT
+                                )
+                                    .show()
+                                showDialog = DialogType.NONE
+                            },
+                            onDismiss = {
+                                showDialog = DialogType.NONE
+                            },
+                        )
+                    }
+
+                    DialogType.NONE -> {}
+                }
+                SearchField(
+                    searchQuery = searchQuery,
+                    chipState = chipState,
+                    onValueChange = { searchQuery = it },
+                )
+
+                LazyColumn {// selected users
+                    items(users.filter { uiState.selectedUserIds.contains(it.documentId) }) { user ->
+                        UserRow(
+                            user,
+                            true,
+                        ) { _ ->
+                            viewModel.userSelectionClicked(user)
+                        }
+                    }
+                }
+                HorizontalDivider(modifier = Modifier.padding(8.dp))
+                if (filteredUnselectedUsers.isNotEmpty()) {
+                    LazyColumn {// all users
+                        items(filteredUnselectedUsers) { user ->
+                            UserRow(
+                                user,
+                                false,
+                            ) { _ ->
+                                viewModel.userSelectionClicked(user)
+                            }
+                        }
+                    }
+                } else {
+                    Text(
+                        text = "Nincsenek a feltételeknek megfelelő tagok",
+                        modifier = Modifier.padding(16.dp),
+                        style = MaterialTheme.typography.headlineSmall,
+                        textAlign = androidx.compose.ui.text.style.TextAlign.Center
+                    )
+                }
+            }
+        }
+    }
+
+    @Composable
+    fun UserRow(
```

Comment:
A Composable dolgokban az a legszebb és attól lesznek könnyen érthetőek, hogy funkcionálisak, és ezért jobb, ha nem metódusok, hanem "standalone" függvények. Ha függnek valamitől ami az osztályban van, akkor adjuk át azt is, de maga a fv. legyen "pure function".